app/code/community/Bolt/Boltpay/Model/BoltOrder.php

```diff
@@ -225,13 +206,26 @@ function ($item) use ($quote, &$calculatedTotal, $boltHelper) {
             ////////////////////////////////////////////////////////////////////////////////////
             // For one page checkout type include tax and shipment / address data in submission.
             ////////////////////////////////////////////////////////////////////////////////////
-            $cartSubmissionData['total_amount'] = round($totals["grand_total"]->getValue() * 100);
 
-            if (@$totals['tax']) {
-                $cartSubmissionData['tax_amount'] = round($totals['tax']->getValue() * 100);
-                $calculatedTotal += $cartSubmissionData['tax_amount'];
+
+            if (Mage::app()->getStore()->isAdmin()) {
+                /* @var Mage_Adminhtml_Block_Sales_Order_Create_Totals $totalsBlock */
+                $totalsBlock =  Mage::app()->getLayout()->createBlock("adminhtml/sales_order_create_totals_shipping");
+
+                /* @var Mage_Sales_Model_Quote_Address_Total $grandTotal */
+                $grandTotal = $totalsBlock->getTotals()['grand_total'];
+                $cartSubmissionData['total_amount'] = (int) round($grandTotal->getValue() * 100);
+
+                /* @var Mage_Sales_Model_Quote_Address_Total $taxTotal */
+                $taxTotal = $totalsBlock->getTotals()['tax'];
+                $totalTax = $this->addTaxForAdmin($taxTotal, $cartSubmissionData);
+            } else {
+                $cartSubmissionData['total_amount'] = round($totals["grand_total"]->getValue() * 100);
+                $totalTax = $this->addTax($totals, $cartSubmissionData);
```

Comment:
I don't like method with side effect (mutating object passed as argument). can we instead do

```
$totalTax =  $this->getTax($totals);
$cartSubmissionData['tax_amount'] = $totalTax;
```

Reply:
No.  The point is that we need access to override how tax is calculated.  What you wrote above is the simplest case, but if some more complex logic needs to be implemented on how tax is calculated because a plugin is used, we want to be able to manipulate the add tax method.

I will offline you some links.

Reply:
Actually, I re-read the request and understand why in this instance you are saying what you are saying.  This has been updated.